pydexec/command.py

```diff
@@ -18,6 +18,65 @@ def args(self, *args):
         self._args.extend(args)
         return self
 
+    def env(self, env_key, env_val):
+        """
+        Add the environment variable ``env_key`` with the value ``env_val``.
+        """
+        self._env[env_key] = env_val
+        return self
+
+    def env_remove(self, env_key):
+        """ Remove the environment variable ``env_key``. """
+        del self._env[env_key]
+        return self
+
+    def env_clear(self):
+        """ Clear all environment variables. """
+        self._env = {}
+        return self
+
+    def env_arg(self, env_key, default=None, required=False, remove=True):
+        """
+        Add an argument with the value of the ``env_key`` environment variable
+        if it is set.
+
+        :param env_key: environment variable name to use
+        :param default: default value if the environment variable is not set
+        :param required: raise an error if the environment variable is not set
+        :param remove: remove the variable from the command's environment
+        """
+        env_op = self._env.pop if remove else self._env.get
+        env_val = env_op(env_key, default)
+        if env_val is not None:
+            self._args.append(env_val)
+        elif required:
+            raise RuntimeError(
+                'Environment variable "%s" is required to determine an '
+                'argument for program "%s"' % (env_key, self._program))
+        return self
+
+    def env_opt(self, opt_key, env_key, default=None, required=False,
+                remove=True):
+        """
+        Add the option ``opt_key`` with the value of the ``env_key``
+        environment variable if it is set.
+
+        :param opt_key: the option to set (e.g. ``--volume``)
+        :param env_key: environment variable name to use
+        :param default: default value if the environment variable is not set
+        :param required: raise an error if the environment variable is not set
+        :param remove: remove the variable from the command's environment
+        """
+        env_op = self._env.pop if remove else self._env.get
+        env_val = env_op(env_key, default)
+        if env_val is not None:
+            self._args.extend([opt_key, env_val])
+        elif required:
+            raise RuntimeError(
+                'Environment variable "%s" is required to determine option '
+                '"%s" for program "%s"' % (env_key, opt_key, self._program))
+        return self
```

Comment:
Can `env_arg()` and `env_opt()` not perhaps share some of their logic?

Maybe the "get a value" part can be factored out and the "add to args" part can be kept separate.